framework/CMakeLists.txt

```diff
@@ -285,38 +291,38 @@ set(ANDROID_FILES
 
 set(WINDOWS_FILES
     # Header Files
-    platform/windows/windows_platform.h
+    platform/desktop/windows/windows_platform.h
     # Source Files
-    platform/windows/windows_platform.cpp)
+    platform/desktop/windows/windows_platform.cpp)
 
 set(UNIX_FILES
     # Header Files
-    platform/unix/unix_platform.h
+    platform/desktop/unix/unix_platform.h
     # Source Files
-    platform/unix/unix_platform.cpp)
+    platform/desktop/unix/unix_platform.cpp)
 
 set(GLFW_FILES
     # Header Files
-    platform/glfw_window.h
+    platform/desktop/glfw_window.h
     # Source Files
-    platform/glfw_window.cpp)
+    platform/desktop/glfw_window.cpp)
 
 set(LINUX_D2D_FILES
     # Header Files
-    platform/unix/unix_d2d_platform.h
-    platform/unix/direct_window.h
+    platform/desktop/unix/unix_d2d_platform.h
+    platform/desktop/unix/direct_window.h
     # Source Files
-    platform/unix/unix_d2d_platform.cpp
-    platform/unix/direct_window.cpp) 
+    platform/desktop/unix/unix_d2d_platform.cpp
```

Comment:
It looks as though you've categorised all unix platforms as desktop, which isn't the case.
Our STB platforms are embedded unix - very much not 'desktop'.

Reply:
I had a think about this, it makes sense to leave them separate but I still wanted to remove the duplication. The platform now hosts some default functions and I had to edit the android platforms prepare call to remove the asynchronous step and have the same initialize -> prepare deterministic lifecycle for all platforms.

@SaschaWillems sorry this might need a reapproval